test_suite/tests/ui/fail_with_invalid_codec_attrs.stderr

```diff
@@ -16,4 +16,4 @@ error: proc-macro derive panicked
 11 | #[derive(TypeInfo, Encode)]
    |          ^^^^^^^^
    |
-   = help: message: Internal error, parse_meta must have been checked: Error("expected literal")
+   = help: message: Internal error. `#[codec(…)]` attributes must be checked in `parity-scale-codec`. This is a bug.: Error("expected literal")
```

Comment:
I still don't 100% get this - the errors above were generated by `parity-scale-codec` so the check appears to have happened correctly - so this error is not necessary?

Reply:
Right, it’s there to chastise sloppy developers and should never reach users. It means that the attribute syntax checks never happened or are buggy; the derive author has to explicitly make a method call so there’s nothing stopping human error. 
We could unwrap here but that’s not better. 
The trybuild is here as a guard against buggy parity-scale-codec releases.

Reply:
>  should never reach users

Forgive me :see_no_evil: , but doesn't this trybuild demonstrate that this error does in fact reach the user even though the syntax checks did happen?

Reply:
You are right and I am wrong. First of all I'm heaping on a bunch of different syntax errors in the same `enum` which makes it hard to understand what is going on. If I use a single-syntax error at a time it's easier to follow.

```rust
#[derive(Encode, TypeInfo)]
enum EnumAttrsValidation {
    #[codec(index = a)]
    Thing(u32),
}
```

Gives this error:

```
error: proc-macro derive panicked
  --> $DIR/fail_with_invalid_codec_attrs.rs:17:18
   |
17 | #[derive(Encode, TypeInfo)]
   |                  ^^^^^^^^
   |
   = help: message: Internal error. `#[codec(…)]` attributes must be checked in `parity-scale-codec`. This is a bug.: Error("expected literal")

error: expected literal
  --> $DIR/fail_with_invalid_codec_attrs.rs:19:21
   |
19 |     #[codec(index = a)]
   |                     ^
```

So `parity-scale-codec` is giving the proper error message back, whereas `scale-info` just panics. What I was trying to say with this test is that `scale-info` will not try to check attributes from `parity-scale-codec`. 

We have no way of knowing if the `Encode` macro ran or how it went, so we can either panic or re-implement their attribute checking and print the proper error twice.

I think we should panic but my error message must change. How about: `"Bad index in `#[codec(index = …)]`: Error("expected literal")`.

Wdyt?

Reply:
Cool, maybe could mention somehow to see the other error generated by `parity-scale-codec`.